src/Zip/FileSelector.cs

```diff
@@ -1165,6 +1165,10 @@ private static SelectionCriterion _ParseCriterion(String s)
                             // if (m.StartsWith("'"))
                             //     m = m.Replace("\u0006", " ");
 
+                            //Fix for Unix -> NormalizeCriteriaExpression replaces all slashes with backslashes
+                            if (Path.DirectorySeparatorChar == '/')

```

Comment:
NormalizeCriteriaExpression is used in many places (not only filename) and there are probably scenarios where slashes should get replaced by backslashes